app/services/user_destroyer.rb

```diff
@@ -21,6 +21,8 @@ def destroy(user, opts = {})
     # default to using a transaction
     opts[:transaction] = true if opts[:transaction] != false
 
+    prepare_for_destroy(user) if opts[:prepare_for_destroy] == true
```

Comment:
If this happens outside the transaction, what would happen if the transaction fails?

Reply:
This is fine, if it halfway fails, user can still be deleted, so all that will happen is you will have a half deleted user when you wanted a full deleted user